AI #1277: Create Conformance Rules for ChargePeriodStart

TechinMama opened the pull request:
### Summary: PR for Conformance definition for column: chargeperiodstart #1277 

## Changes Made

- Updated json values for the chargeperiodstart file based on the chargeperiodestart_cr.md file
- Added the json output to the ../conformance_rules/columns/ directory
- Updated the conformance_datasets.json file based on alphabetic ordering. 

## Notes:
- Validation for Requirements in "ChargePeriodStart-C-005-M" section of the file. 
- This commit includes references to ChargePeriodEnd" in PR 1398 but should not.

TechinMama committed:
conformance rule json and dataset additions

TechinMama committed:
chargeperiod json updates to checkfunction

TechinMama committed:
Merge branch '1121-ai-align-on-approach-for-scrs' into 1278-cr-ChargePeriodEnd-1278

TechinMama committed:
chargeperiodend json updates

TechinMama committed:
Merge branch '1278-cr-ChargePeriodEnd-1278' of https://github.com/A-M-TechConsulting/FOCUS_Spec into 1278-cr-ChargePeriodEnd-1278

TechinMama committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

removing commits from previous changes

# This is the commit message #2:

updated chargestart file

TechinMama committed:
removing commits from previous changes

mike-finopsorg commented on specification/conformance/conformance_rules/columns/chargeperiodend.json:
```diff
@@ -0,0 +1,146 @@
+{
+    "ChargePeriodEnd-C-000-M": {
+        "Function": "Composite",
+        "Reference": "Charge Period End",
```
```suggestion
        "Reference": "ChargePeriodEnd",
```

mike-finopsorg commented on specification/conformance/conformance_rules/columns/chargeperiodend.json:
```diff
@@ -0,0 +1,146 @@
+        "Reference": "Charge Period End",
+        "EntityType": "Column",
+        "ApplicabilityCriteria": [
+            "All Rows"
```
```suggestion
        "ApplicabilityCriteria": [],
```

mike-finopsorg commented on specification/conformance/conformance_rules/columns/chargeperiodend.json:
```diff
@@ -0,0 +1,146 @@
+        "Function": "Validation",
+        "Reference": "Charge Period End",
+        "EntityType": "Column",
+        "ApplicabilityCriteria": [
+            "All Rows"
+        ],
```
```suggestion
        "ApplicabilityCriteria": [],
```

TechinMama committed:
Update specification/conformance/conformance_rules/columns/chargeperiodend.json

camel casing

Co-authored-by: Mike Fuller <mike@finops.org>

TechinMama committed:
Update specification/conformance/conformance_rules/columns/chargeperiodend.json

empty list

Co-authored-by: Mike Fuller <mike@finops.org>

TechinMama committed:
Update specification/conformance/conformance_rules/columns/chargeperiodend.json

empty list

Co-authored-by: Mike Fuller <mike@finops.org>

caligo-erik commented on specification/conformance/conformance_rules/columns/chargeperiodstart.json:
```diff
@@ -0,0 +1,154 @@
+        },
+        "CRVersionIntroduced": "1.2",
+        "Status": "Active",
+        "Notes": "Cross-column reference: DateTimeFormat"
```
DateTimeFormat sounds like a format, and less like a reference to another column.

TechinMama replied:
Hi @caligo-erik That's in the notes section. What is your suggestion or review here? I believe thats an attribute, but based on other entries we can likely just remove it. Thanks so much!

TechinMama committed:
updated chargestart file

TechinMama committed:
Merge branch '1277-cr-ChargePeriodStart-1277' of https://github.com/A-M-TechConsulting/FOCUS_Spec into 1277-cr-ChargePeriodStart-1277

TechinMama committed:
removing commits from previous changes

updated chargestart file

Update specification/conformance/conformance_rules/columns/chargeperiodend.json

camel casing

Co-authored-by: Mike Fuller <mike@finops.org>

Update specification/conformance/conformance_rules/columns/chargeperiodend.json

empty list

Co-authored-by: Mike Fuller <mike@finops.org>

Update specification/conformance/conformance_rules/columns/chargeperiodend.json

empty list

Co-authored-by: Mike Fuller <mike@finops.org>

TechinMama committed:
Merge branch '1277-cr-ChargePeriodStart-1277' of https://github.com/A-M-TechConsulting/FOCUS_Spec into 1277-cr-ChargePeriodStart-1277

mike-finopsorg commented:
Manually imported into the base branch and updated to pass unit tests thanks
